states/etcd/remove/binlog.go

```diff
@@ -0,0 +1,167 @@
+package remove
+
+import (
+	"context"
+	"fmt"
+	"path"
+	"time"
+
+	"github.com/spf13/cobra"
+	clientv3 "go.etcd.io/etcd/client/v3"
+)
+
+var backupKeyPrefix = "birdwatcher/backup"
+
+// BinlogCommand returns remove binlog file from segment command.
+func BinlogCommand(cli clientv3.KV, basePath string) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "binlog",
+		Short: "Remove binlog file from segment with specified segment id and binlog key",
+		Run: func(cmd *cobra.Command, args []string) {
+			logType, err := cmd.Flags().GetString("logType")
+			if err != nil {
+				fmt.Println(err.Error())
+				return
+			}
+
+			collectionID, err := cmd.Flags().GetInt64("collectionID")
+			if err != nil {
+				fmt.Println(err.Error())
+				return
+			}
+
+			partitionID, err := cmd.Flags().GetInt64("partitionID")
+			if err != nil {
+				fmt.Println(err.Error())
+				return
+			}
+
+			segmentID, err := cmd.Flags().GetInt64("segmentID")
+			if err != nil {
+				fmt.Println(err.Error())
+				return
+			}
+
+			fieldID, err := cmd.Flags().GetInt64("fieldID")
+			if err != nil {
+				fmt.Println(err.Error())
+				return
+			}
+
+			var key string
+			switch logType {
+			case "binlog":
+				key = path.Join(basePath, "datacoord-meta",
+					fmt.Sprintf("binlog/%d/%d/%d/%d", collectionID, partitionID, segmentID, fieldID))
+			case "deltalog":
+				key = path.Join(basePath, "datacoord-meta",
+					fmt.Sprintf("deltalog/%d/%d/%d/%d", collectionID, partitionID, segmentID, fieldID))
+			case "statslog":
+				key = path.Join(basePath, "datacoord-meta",
+					fmt.Sprintf("statslog/%d/%d/%d/%d", collectionID, partitionID, segmentID, fieldID))
+			default:
+				fmt.Println("logType unknown:", logType)
+			}
+
+			restore, err := cmd.Flags().GetBool("restore")
+			if err != nil {
+				fmt.Println(err.Error())
+				return
+			}
+			if restore {
+				err = restoreBinlog(cli, key)
+				if err != nil {
+					fmt.Println(err.Error())
+				}
+				return
+			}
+
+			err = backupBinlog(cli, key)
+			if err != nil {
+				fmt.Println(err.Error())
+				return
+			}
```

Comment:
maybe put this step after `do` flag check? otherwise it will backup multiple times despite of the value of `do` flag value